pkg/rancher/rancher.go

```diff
@@ -210,6 +210,7 @@ func New(ctx context.Context, clientConfg clientcmd.ClientConfig, opts *Options)
 		Handler: responsewriter.Chain{
 			auth.SetXAPICattleAuthHeader,
 			responsewriter.ContentTypeOptions,
+			responsewriter.NoCache,
```

Comment:
One question here:

IIRC the original resource that this was requested for was the nodeplan resources. Near as I can tell, NodePlan is a field of the Node object (so really you would use the node endpoint to get the node and then get the plan from that).

Norman has a customization in place for the node objects which sets cache-control to private - (https://github.com/rancher/rancher/blob/649e3345fcb9c913b1dd2c056ac9cf8ade1ac3e9/pkg/api/norman/customization/node/node.go#L268). Would we want to change this?

Notes on the above:
- That only seems to happen for the links field, so the user would have to do something that interacted with the link to get that override value. I think that this is unlikely (basic get to this endpoint didn't have the header).
- Since the header is set to `private`, I think that it is unlikely that this would cause cache issues (from what I was reading, CDNs and the like would not cache the nodeplan's even if the node response came from this endpoint - it would only be cached by the browser).

All that being said, I'm wondering if the above is something that we should change.

Reply:
The node config/plan is retrieved via the endpoint `/v3/connect/config`, not the node endpoint. In addition, as you pointed out, this is unlikely to cause the caching problem that we are trying to fix here.

Reply:
Thanks for pointing that out. Since this appears to be active before the MCM routes (which is where this occurs) and since that route seems to not override the header, I'm good here.